### DISCUSSION TITLE
nested for loop

### DISCUSSION THREAD

**Discussion by ckkkcsg** [cue-lang/cue#1379] [Q&A] [answered]
const label_traitdef = `
    #EndpointSubset: {
      address?: [...#EndpointAddress]
      ports?: [...#EndpointPort]
    }
    #EndpointAddress: {
      ip: string
      hostname?: string
      NodeName?: string
      targetRef?: #TargetRef
    }
    #EndpointPort: {
      name?: string
      port: int
      protocol?: "TCP" | "UDP" | "SCTP"
    }
    #TargetRef: {
      apiVersion: string
      kind: string
    }

	#Endpoints: [...{
		ip: string
	}]
    
    parameter: {
      endpointSubset?: [...#EndpointSubset]
    }

    output: {
		//endpoints: parameter.endpointSubset[0].address[0]
		for subset in parameter.endpointSubset {
			endpoints: #Endpoints & [
				for addr in subset.address {
					for po in subset.ports {
						ip: addr.ip + "\(po.port)"
					}
				}
			]
		}
		
    }
`

const param = `
parameter: {
	endpointSubset: [
		{
			address: [
				{
					ip: "10.244.0.100"
				},
				{
					ip: "10.244.154.81"
				},
			]
			ports: [
				{
					port: 9300
				},
				{
					port: 9200
				},
			]
		},
	]
}
`

func main() {
	ctx := cuecontext.New()

	value := ctx.CompileString(label_traitdef)
	if value.Err() != nil {
		fmt.Println("load error:", value.Err())
		return
	}

	fmt.Println("err:", value.Validate())
	//fmt.Println(value)

	val := ctx.CompileString(param)
	if val.Err() != nil {
		fmt.Println("val load fail:", val.Err().Error())
		return
	}
}

i got val load fail: output.endpoints.0.ip: conflicting values "10.244.0.1009200" and "10.244.0.1009300"

**Comment by verdverm** [comment, marked as answer]
This is expected, see #1375

**Comment by ckkkcsg** [comment]
> This is expected, see #1375

i want got
endpoints:[ { ip: 10.244.0.100:9200 }, { ip: 10.244.0.100:9300 }, { ip: 10.244.154.81:9200 }, { ip: 10.244.154.81:9300 } ]

what should I do？

**Comment by verdverm** [comment]
I'd probably comprehend nested lists and then use list.FlattenN
